connectors/utils.py

```diff
@@ -435,6 +436,24 @@ async def apply_retry_strategy(strategy, interval, retry):
             raise UnknownRetryStrategyError()
 
 
+def with_concurrency_control(retries=3):
+    def wrapper(func):
+        @functools.wraps(func)
+        async def wrapped(*args, **kwargs):
+            retry = 1
+            while retry <= retries:
+                try:
+                    return await func(*args, **kwargs)
+                except ConflictError as e:
+                    if retry >= retries:
+                        raise e
```

Comment:
I suggest adding a log line (maybe at DEBUG?) to show that there was a conflict and that we're executing a retry